systemtests/src/main/java/io/enmasse/systemtest/manager/SharedIoTManager.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.manager;
+
+import io.enmasse.address.model.AddressSpace;
+import io.enmasse.iot.model.v1.IoTConfig;
+import io.enmasse.iot.model.v1.IoTConfigBuilder;
+import io.enmasse.iot.model.v1.IoTProject;
+import io.enmasse.systemtest.Environment;
+import io.enmasse.systemtest.UserCredentials;
+import io.enmasse.systemtest.amqp.AmqpClient;
+import io.enmasse.systemtest.amqp.AmqpClientFactory;
+import io.enmasse.systemtest.certs.CertBundle;
+import io.enmasse.systemtest.logs.CustomLogger;
+import io.enmasse.systemtest.mqtt.MqttClientFactory;
+import io.enmasse.systemtest.platform.apps.SystemtestsKubernetesApps;
+import io.enmasse.systemtest.utils.CertificateUtils;
+import io.enmasse.systemtest.utils.IoTUtils;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+import static io.enmasse.systemtest.bases.iot.ITestIoTBase.iotProjectNamespace;
+import static io.enmasse.systemtest.iot.DefaultDeviceRegistry.deviceRegistry;
+import static io.enmasse.systemtest.utils.IoTUtils.createIoTConfig;
+import static io.enmasse.systemtest.utils.IoTUtils.createIoTProject;
+
+
+public class SharedIoTManager extends ResourceManager {
+
+    private static final Logger LOGGER = CustomLogger.getLogger();
+    private static SharedIoTManager instance;
+    protected AmqpClientFactory amqpClientFactory = null;
+    protected MqttClientFactory mqttClientFactory = null;
+    private IoTProject sharedIoTProject = null;
+    private IoTConfig sharedIoTConfig = null;
+    private AmqpClient amqpClient;
+
+
+    public static synchronized SharedIoTManager getInstance() {
```

Comment:
I think you should make the constructor private, and this class final, if you want to enforce your singleton pattern.